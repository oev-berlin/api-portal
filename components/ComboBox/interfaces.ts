export interface ComboBoxProps{
    searchOptions: string[],
}
