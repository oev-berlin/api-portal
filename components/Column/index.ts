import { Column } from './Column';
import { ColumnProps } from './interfaces';

export { Column };
export type { ColumnProps };
