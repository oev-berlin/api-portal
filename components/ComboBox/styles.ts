import styled from 'styled-components';
import { Autocomplete } from '@mui/material';

export const StyledAutocomplete = styled(Autocomplete)`
  width: 300px;
  margin-right: auto;
  margin-left: auto;
`;
