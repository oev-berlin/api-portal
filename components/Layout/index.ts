import { Layout } from './Layout';
import { LayoutProps } from './interfaces';

export { Layout };
export type { LayoutProps };
