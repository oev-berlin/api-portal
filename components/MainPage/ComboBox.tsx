import {Autocomplete, TextField} from "@mui/material";
import {projects} from "./Data";

export default function ComboBox (options:string[]){
    return (
        <Autocomplete
            disablePortal
            id="combo-box-demo"
            options={options}
            sx={{maxWidth:0.5, mx: "auto", my:4}}
            renderInput={(params) => <TextField {...params} label="Project"/>}
        />);
}
