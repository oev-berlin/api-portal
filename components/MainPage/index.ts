export {CustomisedCard} from "./ProjectCard";
export { MainPage } from './MainPage';
