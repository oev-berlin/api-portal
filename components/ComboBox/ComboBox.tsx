import React from 'react';
import { ListItem, TextField } from '@mui/material';

import { useRouter } from 'next/router';
import { ComboBoxProps } from './interfaces';
import { StyledAutocomplete } from './styles';

export default function ComboBox({ searchOptions }: ComboBoxProps) {
  const router = useRouter();
  function handelClick(projectTitle:string) {
    router.push(`/project/${projectTitle}`);
  }

  return (
    <StyledAutocomplete
      disablePortal
      options={searchOptions}
      renderOption={(props, option) => {
        const projectTitle: string = option as string;
        return (
          <ListItem
            {...props}
            key={projectTitle}
            onClick={() => handelClick(projectTitle)}
          >{projectTitle}
          </ListItem>
        );
      }}
      renderInput={(props) => <TextField {...props} label="Project" />}
    />
  );
}
