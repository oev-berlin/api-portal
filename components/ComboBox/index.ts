import ComboBox from './ComboBox';
import { ComboBoxProps } from './interfaces';

export { ComboBox };
export type { ComboBoxProps };
