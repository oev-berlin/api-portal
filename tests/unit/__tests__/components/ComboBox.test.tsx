import { render, cleanup, waitFor } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import React from 'react';
import { ComboBox, ComboBoxProps } from '../../../../components/ComboBox';
import 'jest-styled-components';

expect.extend(toHaveNoViolations);

const setupComponent = ({ projectsTitles }: ComboBoxProps) => render(
  <ComboBox
    searchOptions={projectsTitles}
  />,
);

describe('ComboBox', () => {
  afterEach(() => {
    cleanup();
  });

  describe('UI Tests', () => {
    it('ComboBox component should render without crashing', async () => {
      const screen = setupComponent({
        searchOptions: ['salneo', 'tidy'],
      });
      expect(screen.container).toBeDefined();
    });
    it('ComboBox component (searchOptions is empty) should render without crashing', async () => {
      const screen = setupComponent({
        searchOptions: [],
      });
      expect(screen.container).toBeDefined();
    });
  });

  describe('Snapshots Tests', () => {
    it('should match a basic snapshot', () => {
      const { container } = setupComponent({ searchOptions: ['salneo', 'tidy'] });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (searchOptions is empty)', () => {
      const { container } = setupComponent({ searchOptions: [] });
      expect(container).toMatchSnapshot();
    });
  });
  describe('Accessibility Tests', () => {
    it('should pass a basic accessibility test', async () => {
      const { container } = setupComponent({ searchOptions: ['salno'] });
      const results = await waitFor(() => axe(container));
      expect(results).toHaveNoViolations();
    });
  });
});
