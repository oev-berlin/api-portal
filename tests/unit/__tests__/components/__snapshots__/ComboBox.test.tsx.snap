// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboBox Snapshots Tests should match a basic snapshot (searchOptions is empty) 1`] = `
.c0 {
  width: 300px;
}

<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon c0 css-1qqsdnr-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="mui-8"
        id="mui-8-label"
      >
        Project
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id="mui-8"
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Project
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`ComboBox Snapshots Tests should match a basic snapshot 1`] = `
.c0 {
  width: 300px;
}

<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon c0 css-1qqsdnr-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="mui-6"
        id="mui-6-label"
      >
        Project
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id="mui-6"
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Project
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
